src/utilities/string_functions.js

```diff
@@ -295,20 +295,23 @@ p5.prototype.nf = function(nums, left, right) {
 };
 
 function doNf(num, left, right) {
+  let isNegative = num < 0;
+  num = Math.abs(num);
   let [leftPart, rightPart] = num.toString().split('.');
 
+
   if (typeof right === 'undefined') {
     leftPart = leftPart.padStart(left, '0');
-    return rightPart ? leftPart + '.' + rightPart : leftPart;
+    let result = rightPart ? leftPart + '.' + rightPart : leftPart;
+    return isNegative ? '-' + result : result;
   } else {
-    let roundedOff = num.toFixed(right);
-    [leftPart, rightPart] = roundedOff.toString().split('.');
+    let roundedNum = Math.round(num * Math.pow(10, right));
+    roundedNum = roundedNum / Math.pow(10, right);
```

Comment:
Can we roundooff the num by simply doing this:
```suggestion
    let roundedNum = num.toFixed(right);
```

Let me know if it works for you @Akhilbisht798 .

Reply:
yes we can, previous commit was using it. but as you have mentioned that 

> For example, if we use let num = nf(123.58, 1,1);, the expected output should be 123.6, as it involves rounding the numbers after the decimal point. However, in your implementation, it currently results in 123.5 without rounding off correctly. Do you think we could adjust this?


This is was not working with `num.toFixed(right);`. But if you want I can change it back.

Reply:
Were you also able to reproduce the same error of rounding off with `num.toFixed(right);`. I am currently testing and here's the build and it works well.

https://editor.p5js.org/aman12345/sketches/Pq9D8HM0h